generators/server/index.js

```diff
@@ -630,18 +674,25 @@ module.exports = JhipsterServerGenerator.extend({
             ];
 
             this.prompt(prompts, function (props) {
-                if (this.applicationType === 'microservice' || this.applicationType === 'gateway') {
-                    this.authenticationType = 'jwt';
-                } else {
-                    this.authenticationType = props.authenticationType;
-                }
+                this.authenticationType = props.authenticationType;
+
                 if (this.authenticationType === 'session') {
                     this.rememberMeKey = crypto.randomBytes(20).toString('hex');
                 }
-                if (this.authenticationType === 'jwt') {
+
+                if (this.authenticationType === 'jwt' || this.authenticationType === 'uaa' || this.applicationType === 'microservice' || this.applicationType === 'uaa') {
                     this.jwtSecretKey = crypto.randomBytes(20).toString('hex');
                 }
 
+                //this now is job for uaa
+                if(this.applicationType === 'gateway' && this.authenticationType === 'uaa') {
+                    this.skipUserManagement = true;
+                }
+
+                if(applicationType === 'uaa') {
```

Comment:
this is not a template you cannot get anything in `applicationType` here it would be un defined. You are accessing `this.applicationType ` a line above